Sources/XMTP/Conversations.swift

```diff
@@ -11,6 +11,61 @@ import XMTPProto
 public struct Conversations {
 	var client: Client
 
+	public func newConversation(with peerAddress: String, context: InvitationV1.Context? = nil) async throws -> Conversation {
+		guard let contact = try await client.getUserContact(peerAddress: peerAddress) else {
+			throw ContactBundleError.notFound
+		}
+
+		// See if we have an existing v1 convo
+		if context?.conversationID == nil || context?.conversationID == "" {
+			let invitationPeers = try await listIntroductionPeers()
+			if let peerSeenAt = invitationPeers[peerAddress] {
+				return .v1(ConversationV1(client: client, peerAddress: peerAddress, sentAt: peerSeenAt))
+			}
+		}
+
+		// If the contact is v1, start a v1 conversation
+		if case .v1 = contact.version, context?.conversationID == nil || context?.conversationID == "" {
+			return .v1(ConversationV1(client: client, peerAddress: peerAddress, sentAt: Date()))
+		}
+
+		// See if we have a v2 conversation
+		for sealedInvitation in try await listInvitations() {
+			if !sealedInvitation.involves(contact) {
+				continue
+			}
+
+			let invite = try sealedInvitation.v1.getInvitation(viewer: client.keys)
+			if invite.context.conversationID == context?.conversationID, invite.context.conversationID != "" {
```

Comment:
I'm a bit curious where `conversationID` gets used? I see the JS SDK checks for it in a bunch of places but I didn't see any places where it gets a real value? Is it something clients are meant to specify?

Reply:
`newConversation`

Reply:
Yeah, [here's an example](https://github.com/lensterxyz/lenster/blob/30d458b792726ff33430ae1012b8385b9f4f656b/apps/web/src/components/utils/hooks/useGetConversation.tsx#L36) of it being used in the wild. It allows consumers of the SDK to do things like filter out all other convos except the ones that match the convo id. In XMTP's example web app, we are acting like a "universal inbox" so we don't do any client-side filtering by convo id there.

Reply:
Gotcha, thanks!